docs/auxil/admonition_inserter.py

```diff
@@ -295,18 +302,20 @@ def _create_shortcuts(self) -> dict[collections.abc.Callable, str]:
         # that this given method is a shortcut for a Bot method
         for class_name, cls in inspect.getmembers(telegram, predicate=inspect.isclass):
 
-            for method_name, method in inspect.getmembers(cls, predicate=inspect.isfunction):
+            # no need to inspect Bot's own methods, as Bot can't have shortcuts in But
+            if cls is telegram.Bot:
+                continue
+
+            for method_name, method in _iter_own_public_methods(cls):
+
+                # .getsourcelines() returns a tuple. Item [1] is an int
+                for line in inspect.getsourcelines(method)[0]:
+                    bot_method_match = bot_method_pattern.search(line)
```

Comment:
```suggestion
                    if not bot_method_match := bot_method_pattern.search(line):
                        continue
```

since running this requires py3.9+ now anyway :)

Reply:
so you think an occasional regex might be too complex but the walrus operator that no-one ever learned is not confusing 😆 

because of some precedence issues this had to be wrapped in another set of parentheses

Reply:
walrus is way less confusing than regex ;P